src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/ServerVariables.cs

```diff
@@ -55,7 +55,7 @@ public static PatternSegment FindServerVariable(string serverVariable)
                 case "REQUEST_URI":
                     return new UrlSegment();
                 default:
-                    throw new FormatException("Unrecognized server variable");
+                    throw new FormatException(Resources.FormatError_InputParserUnrecognizedParameter(serverVariable, context.Index));
```

Comment:
The code has become inconsistent on strings vs resources. We should add an issue to clean this up. (You don't need to add to this PR.)